sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeColumnCommand.scala

```diff
@@ -50,7 +52,26 @@ case class AnalyzeColumnCommand(
     val sizeInBytes = CommandUtils.calculateTotalSize(sparkSession, tableMeta)
 
     // Compute stats for each column
-    val (rowCount, newColStats) = computeColumnStats(sparkSession, tableIdentWithDB, columnNames)
+    val conf = sparkSession.sessionState.conf
+    val relation = sparkSession.table(tableIdent).logicalPlan
+    val attributesToAnalyze = if (allColumns) {
+      relation.output
```

Comment:
Are we still able to create a table with zero column? for example, using dataframewriter?

Reply:
@gatorsmile I think we don't allow it any more. Please see [test](https://github.com/dilipbiswal/spark/blob/7920b29f667de38ca5222e4de8f62eea688d67e4/sql/core/src/test/scala/org/apache/spark/sql/FileBasedDataSourceSuite.scala#L111:L136). Please let me know what you think..